build.gradle

```diff
@@ -45,15 +45,6 @@ plugins {
 try {
   // apply these plugins in a try-catch block so that we can handle cases without .git directory
   apply plugin: 'com.palantir.git-version'
-  apply plugin: 'com.gorylenko.gradle-git-properties'
-  // git properties file for the root project for adding to the source tarball
-  gitProperties {
-    gitPropertiesName = 'iceberg-build.properties'
-    gitPropertiesResourceDir = file("${rootDir}/build")
-    extProperty = 'gitProps'
-    failOnNoGitDirectory = true
-  }
-  generateGitProperties.outputs.upToDateWhen { false }
```

Comment:
This is moved out of the try/catch block because it is safe to add the plugin when `.git` is not present. The `fialOnNoGitDirectory` flag determines whether `generateGitProperties` will succeed if it isn't present. It is set to true here because the build should fail if there is no root `iceberg-build.properties` file and `.git` is not present.

Reply:
👍  Good find. 

When I tried cloning with depth=1 and then `rm -rf .git/` tasks failed, but glad to see you were able to make it work.